File: String/2273. Find Resultant Array After Removing Anagrams.cpp
class Solution {
public:
    vector<string> removeAnagrams(vector<string>& words) {
        for (int i = 1; i < words.size(); i++) {
            string temp1 = words[i];     // Current word
            sort(temp1.begin(), temp1.end()); // Sort the current word

            string temp2 = words[i - 1]; // Previous word
            sort(temp2.begin(), temp2.end()); // Sort the previous word

            // If the sorted words are the same, they are anagrams
            if (temp1 == temp2) {
                words.erase(words.begin() + i); // Remove the current word
                i--; // Adjust the index to account for the removed word
            }
        }
        return words; // Return the updated list
    }
};

File: String/2273. Find Resultant Array After Removing Anagrams.java
class Solution {
    public List<String> removeAnagrams(String[] words) {
        // Create a list to store the final result (non-consecutive anagrams)
        List<String> result = new ArrayList<>();

        // Always add the first word to the result list
        result.add(words[0]);

        // Convert the first word into a sorted version (to compare anagrams)
        char[] firstWordChars = words[0].toCharArray();
        Arrays.sort(firstWordChars);
        String previousSorted = new String(firstWordChars); // stores sorted form of previous word

        // Start checking from the second word onwards
        for (int i = 1; i < words.length; i++) {
            // Convert the current word to a sorted version
            char[] currentWordChars = words[i].toCharArray();
            Arrays.sort(currentWordChars);
            String currentSorted = new String(currentWordChars);

            // If the current sorted word equals the previous sorted word,
            // it means both are anagrams → skip adding to the result
            if (currentSorted.equals(previousSorted)) {
                continue;
            } 
            // Otherwise, add it to the result and update the "previousSorted" pattern
            else {
                result.add(words[i]);
                previousSorted = currentSorted;
            }
        }

        // Return the final list without consecutive anagrams
        return result;
    }
}
